Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -1175,6 +1195,8 @@ static void UpdateBATs(BatTable& bat_table, u32 base_spr)
           valid_bit = 0x3;
         else if ((address >> 28) == 0xE && (address < (0xE0000000 + Memory::L1_CACHE_SIZE)))
           valid_bit = 0x3;
+        if (overlaps_memcheck(address | ((1 << BAT_INDEX_SHIFT) - 1)))
```

Comment:
You might also do the same check with "(batu.BEPI | j) << BAT_INDEX_SHIFT" as this is the virtual address